### ISSUE TITLE
EQUALP should use EQ on objects

### ISSUE THREAD

**Issue by stassats** [drmeister/clasp#227]
```
(progn (defclass foo () ()) (equalp (make-instance 'foo) (make-instance 'foo)))
```

=>
T

**Comment by Shinmera** [comment]
99669d674a6f2b7ce0a9127cf6537fc03066baa5

**Comment by Bike** [comment]
This is no longer happening as of bbe56683b. However I'm suspicious of the fix, because we still have an `Instance_O::equalp` method that should not exist. Gonna try deleting that...

**Comment by Bike** [comment]
Checked closer, we check for structure-object-ness like so https://github.com/drmeister/clasp/blob/bd52cd276fd6062f538e8d7fb79b0e973c9ea5b9/src/core/instance.cc#L346-L348. Works well enough. Probably fixed by bd52cd27
